app/models/spree/tax_rate.rb

```diff
@@ -110,7 +110,7 @@ def compute_amount(item)
             "Notice: Tax refund should not be possible, please check the default zone and " \
             "the tax rate zone configuration"
           ) do |payload|
-            payload.add_metadata :order_tax_zone, order.tax_zone
+            payload.add_metadata :order_tax_zone, item.order.tax_zone
             payload.add_metadata :tax_rate_zone, zone
             payload.add_metadata :default_zone, Zone.default_tax
           end
```

Comment:
If we don't support this code path, shouldn't we also raise an application exception, and remove the next couple of lines? Or are we not sure enough about it to change the behaviour just yet..?

Sure, we don't want more snails, but snails are better than hidden problems. Instance managers are not going to be seeing the bugsnag error so won't know if it's happening unless someone notices and tells them.

Reply:
> Or are we not sure enough about it to change the behaviour just yet..?

My idea was to add Bugsnag to confirm it's not happening in production, I don't feel confident enough yet to remove the code. We should be able to remove the code if we don't see the snails after a month or two.  

I am not sure about raising an exception, it would probably block placing an order if it gets triggered. I guess we could remove "the refund" and just to a normal compute, that would avoid the wrong tax rate. What do you  think ?

Reply:
Thats right, I thought it would be better to block the order, rather than have an order with wrong details.

But I'm not sure I fully understand the context, so am ok with just adding Bugsnag for now, if it helps gain understanding of what's happening.